Source/DafnyDriver/CliCompilation.cs

```diff
@@ -0,0 +1,229 @@
+  }
+
+  public async Task VerifyAllAndPrintSummary() {
+    var statSum = new PipelineStatistics();
+
+    var canVerifyResults = new Dictionary<ICanVerify, CliCanVerifyResults>();
+    Compilation.Updates.Subscribe(ev => {
+
+      if (ev is CanVerifyPartsIdentified canVerifyPartsIdentified) {
+        var canVerifyResult = canVerifyResults[canVerifyPartsIdentified.CanVerify];
+        foreach (var part in canVerifyPartsIdentified.Parts) {
+          canVerifyResult.Tasks.Add(part);
+        }
+
+        if (canVerifyResult.CompletedParts.Count == canVerifyResult.Tasks.Count) {
+          canVerifyResult.Finished.SetResult();
+        }
+      }
+
+      if (ev is BoogieException boogieException) {
+        var canVerifyResult = canVerifyResults[boogieException.CanVerify];
+        canVerifyResult.Finished.SetException(boogieException.Exception);
+      }
+
+      if (ev is BoogieUpdate boogieUpdate) {
+        if (boogieUpdate.BoogieStatus is Completed completed) {
+          var canVerifyResult = canVerifyResults[boogieUpdate.CanVerify];
+          canVerifyResult.CompletedParts.Add((boogieUpdate.ImplementationTask, completed));
+
+          switch (completed.Result.Outcome) {
+            case ConditionGeneration.Outcome.Correct:
+            case ConditionGeneration.Outcome.ReachedBound:
+              Interlocked.Increment(ref statSum.VerifiedCount);
+              break;
+            case ConditionGeneration.Outcome.Errors:
+              Interlocked.Add(ref statSum.ErrorCount, completed.Result.Errors.Count);
+              break;
+            case ConditionGeneration.Outcome.TimedOut:
+              Interlocked.Increment(ref statSum.TimeoutCount);
+              break;
+            case ConditionGeneration.Outcome.OutOfMemory:
+              Interlocked.Increment(ref statSum.OutOfMemoryCount);
+              break;
+            case ConditionGeneration.Outcome.OutOfResource:
+              Interlocked.Increment(ref statSum.OutOfResourceCount);
+              break;
+            case ConditionGeneration.Outcome.Inconclusive:
+              Interlocked.Increment(ref statSum.InconclusiveCount);
+              break;
+            default:
+              throw new ArgumentOutOfRangeException();
+          }
+
+          if (canVerifyResult.CompletedParts.Count == canVerifyResult.Tasks.Count) {
+            canVerifyResult.Finished.TrySetResult();
+          }
+        }
+      }
+    });
+
+    try {
+      var resolution = await Compilation.Resolution;
+      if (errorCount > 0) {
+        return;
+      }
+
+      var canVerifies = resolution.CanVerifies?.DistinctBy(v => v.Tok).ToList();
+
+      if (canVerifies != null) {
+        var orderedCanVerifies = canVerifies.OrderBy(v => v.Tok.pos).ToList();
+        foreach (var canVerify in orderedCanVerifies) {
+          canVerifyResults[canVerify] = new CliCanVerifyResults();
+          await Compilation.VerifyCanVerify(canVerify, false);
+        }
```

Comment:
I have to await before I do the next call, to ensure that the previous call has been added to any queues. Otherwise, the order in which things get verified becomes non-deterministic.